src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -57,11 +107,34 @@ internal ConfigurationClientManager(IList<ConfigurationClientWrapper> clients)
             _clients = clients;
         }
 
-        public IEnumerable<ConfigurationClient> GetAvailableClients(DateTimeOffset time)
+        public async Task<IEnumerable<ConfigurationClient>> GetAvailableClients(DateTimeOffset time, CancellationToken cancellationToken)
         {
-            return _clients.Where(client => client.BackoffEndTime <= time).Select(c => c.Client).ToList();
-        }
+            if (_replicaDiscoveryEnabled)
+            {
+                // The source of truth whether there's avaliable client in the list.
+                int successfulClientsCount = _clients.Where(client => client.FailedAttempts == 0).Count();
+
+                // Refresh fallback clients if:
+                // 1.It's the initial attempt to query available clients on app start-up.
+                // 2.At least one client is available in the last attempt, it's the first attempt that all static and dynamic clients are backed off.
+                // 3.It's not the first attempt that all clients have backed off, as well as it's due time to refresh fallback clients.
+
+                if ((_lastFallbackClientRefresh == default ||
```

Comment:
Should be covered by `time >= _lastFallbackClientRefresh + _fallbackClientRefreshInterval` check.

Reply:
if remove checking `_lastFallbackClientRefresh == default`, it will not perform lookup on app's start-up